bfabric/src/math_helper.py

```diff
@@ -0,0 +1,20 @@
+
+
+def div_int_ceil(n: int, d: int) -> int:
+    """
+    :param n: Numerator
+    :param d: Denominator
+    :return:  Performs integer ceiling division
+    Theoretically equivalent to math.ceil(n/d), but not subject to floating-point errors.
+    """
+    q, r = divmod(n, d)
+    return q + bool(r)
+
+
+
+if __name__ == "__main__":
+    print(
+        div_int_ceil(120, 100),
+        div_int_ceil(200, 100),
+        div_int_ceil(245, 100)
```

Comment:
The major reason for this function is that `math.ceil(n/d)` applies a ceiling function to a floating-point number. There is a risk that ceil(2/4) = 3 instead of 2 because of roundoff errors. I'm happy to also add the test for large numbers, but that was not the reason to introduce it.

Reply:
I'm ok with the function as is, because it's correct and general. But I'm saying it probably is not necessary for our current use case, because, assuming IEEE 754 doubles:

- the concern would apply to cases where the division remainder is 0, because all others are supposed to be rounded-up by `math.ceil` anyhow (ignoring negative numbers for simplicity)
- integer numbers can be [represented exactly](https://en.wikipedia.org/wiki/Double-precision_floating-point_format#Precision_limitations_on_integer_values) as float from -2^53, +2^53 
- (basic operations like `+,-,/,*,sqrt`, i.e. not necessarily `sin(x)` or `pow(a,b)`) floating point operations will be rounded to the nearest exactly representable value (IEEE 754 section 4 intro, section 5)

If you use numbers above the range which is representable correctly, then you might run into problems, which your code solves and the case above would test for.

Reply:
I see, thanks for clarifying. I was not aware roundoff issue was only there for large numbers.